Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptMergeService.java

```diff
@@ -0,0 +1,297 @@
+        } else {
+            // Update
+            ConceptEntry entry = conceptManager.getConceptEntry(conceptsMergeBean.getSelectedConceptId());
+            getConceptEntryFromConceptMergeBean(entry, conceptsMergeBean);
+            conceptManager.storeModifiedConcept(entry, userName);
+        }
+
+    }
+
+    private void deleteMergedConcepts(List<String> conceptIds, String userName, String conceptIdToNotDelete)
+            throws LuceneException, IndexerRunningException {
+        for (String conceptId : conceptIds) {
+            if (!conceptId.equalsIgnoreCase(conceptIdToNotDelete)
+                    && ConceptTypes.SPECIFIC_WORDNET_CONCEPT != conceptTypesService
+                            .getConceptTypeByConceptId(conceptId)) {
+                // Wordnet ids cannot be deleted. So adding the wordnet ids to
+                // the merged concept entry
+                // in getConceptEntryFromConceptMergeBean()
+                conceptManager.deleteConcept(conceptId, userName);
+            }
+        }
+    }
+
+    public void getConceptEntryFromConceptMergeBean(ConceptEntry entry, ConceptsMergeBean conceptMergeBean) {
+        StringBuffer wordBuffer = new StringBuffer();
+        String prefix = "";
+        for (String word : conceptMergeBean.getWord()) {
+            wordBuffer.append(prefix);
+            prefix = ",";
+            wordBuffer.append(word);
+        }
+        entry.setWord(wordBuffer.toString().trim());
+
+        entry.setPos(conceptMergeBean.getSelectedPosValue());
+        entry.setConceptList(conceptMergeBean.getSelectedListName());
+
+        StringBuffer descriptionBuffer = new StringBuffer();
+        prefix = "";
+        for (String description : conceptMergeBean.getDescription()) {
+            descriptionBuffer.append(prefix);
+            prefix = ",";
+            descriptionBuffer.append(description);
+        }
+
+        entry.setDescription(descriptionBuffer.toString().trim());
+
+        StringBuffer synonymBuffer = new StringBuffer();
+        prefix = "";
+        for (String synonymId : conceptMergeBean.getSynonymsids()) {
+            synonymBuffer.append(prefix);
+            prefix = ",";
+            synonymBuffer.append(synonymId);
+        }
+
+        entry.setSynonymIds(synonymBuffer.toString());
+        entry.setTypeId(conceptMergeBean.getSelectedTypeId());
+
+        StringBuffer equalsBuffer = new StringBuffer();
+        prefix = "";
+        for (String equals : conceptMergeBean.getEquals()) {
+            equalsBuffer.append(prefix);
+            prefix = ",";
+            equalsBuffer.append(equals);
+        }
+        entry.setEqualTo(equalsBuffer.toString().trim());
+
+        StringBuffer similarBuffer = new StringBuffer();
+        prefix = "";
+        for (String similarTo : conceptMergeBean.getSimilar()) {
+            similarBuffer.append(prefix);
+            prefix = ",";
+            similarBuffer.append(similarTo);
```

Comment:
instead of all these for loops just use a stringjoiner or even better the collector as shown here:
https://docs.oracle.com/javase/8/docs/api/java/util/StringJoiner.html